component/core/src/main/java/org/exoplatform/social/core/metadata/thumbnail/ImageThumbnailServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+    ThumbnailObject thumbnailObject = new ThumbnailObject(THUMBNAIL_OBJECT_TYPE, Long.toString(fileInfo.getId()));
+    List<MetadataItem> metadataItemList = metadataService.getMetadataItemsByMetadataTypeAndObject(THUMBNAIL_METADATA_TYPE.getName(),
+                                                                                                thumbnailObject);
+    List<MetadataItem> items = metadataItemList.stream()
+                                               .filter(metadataItem -> metadataItem.getProperties() != null
+                                                   && metadataItem.getProperties()
+                                                                  .get(THUMBNAIL_WIDTH_PROPERTY)
+                                                                  .equals(String.valueOf(width))
+                                                   && metadataItem.getProperties()
+                                                                  .get(THUMBNAIL_HEIGHT_PROPERTY)
+                                                                  .equals(String.valueOf(height)))
+                                               .collect(Collectors.toList());
+    if (!items.isEmpty()) {
+      long fileId = Long.parseLong(items.get(0).getParentObjectId());
+      try {
+        return fileService.getFile(fileId);
+      } catch (FileStorageException e) {
+        return null;
+      }
+    } else {
+      byte[] imageContent = imageResizeService.scaleImage(IOUtils.toByteArray(file.getAsStream()), width, height, false, false);
+      FileItem thumbnail = new FileItem(null,
+                                        fileInfo.getName(),
+                                        fileInfo.getMimetype(),
+                                        SOCIAL_NAME_SPACE,
+                                        imageContent.length,
+                                        new Date(),
+                                        identity.getRemoteId(),
+                                        false,
+                                        new ByteArrayInputStream(imageContent));
+      FileItem thumbnailFileItem = fileService.writeFile(thumbnail);
+      FileInfo thumbnailFileInfo = thumbnailFileItem.getFileInfo();
+      ThumbnailObject thumbnailMetadataObject = new ThumbnailObject(THUMBNAIL_OBJECT_TYPE,
+                                                                    Long.toString(fileInfo.getId()),
+                                                                    Long.toString(thumbnailFileInfo.getId()));
+      MetadataKey metadataKey = new MetadataKey(THUMBNAIL_METADATA_TYPE.getName(), THUMBNAIL_METADATA_TYPE.getName(), 0);
+      Map<String, String> properties = new HashMap<>();
+      properties.put(THUMBNAIL_WIDTH_PROPERTY, String.valueOf(width));
+      properties.put(THUMBNAIL_HEIGHT_PROPERTY, String.valueOf(height));
+      metadataService.createMetadataItem(thumbnailMetadataObject, metadataKey, properties, Long.parseLong(identity.getId()));
```

Comment:
```suggestion
      metadataService.createMetadataItem(thumbnailMetadataObject, metadataKey, properties, 0);
```

Reply:
will throw an exception:  https://github.com/Meeds-io/social/blob/develop/component/core/src/main/java/org/exoplatform/social/core/metadata/MetadataServiceImpl.java#L183

Reply:
i need the used id here, so i add the identity as a parameter in the getOrCreateThumbnail? or i add only user identity id?

Reply:
As the thumbnail is related to the original file, should we use the originalFile updater as thumbnail creator ?

Reply:
ok to keep identity as it is the avatar owner identity